### ISSUE TITLE
Feature Suggestion(s): `Manage Games` Improvements

### ISSUE THREAD

**Issue by psygo** [SabakiHQ/Sabaki#599]
Some of my suggestions were discussed in Issue #424. However, that was kind of a confusing discussion, so I will try to start anew.

The improvements I would like to suggest for the `Manage Games` feature are &mdash; I don't know if they are actually possible due to limitations imposed by `sgf` standards:

- **A title for each game in the collection**.
    - This would be very useful for visualizing the collection better. From what I understand, each game inside a collection *can* be named even if they do not represent a separate file, though I'm not 100% sure.
    - As a subfeature, there could also be a toggle to see the games in the collection as a list, where the titles would differentiate each one.
- A **toggle button where you can set the preview to display the beginning of the games or their end**.
    - This topic was originally discussed in Issue #315.
    - I've tried to create a collection of tsumegos recently and the tsumego solutions simply get spoiled by the preview, which is very annoying. This spoiling can even be a nuissance for normal games, because sometimes you don't want to see the final position.
    - The toggle button should probably be on the `top menu bar` and not alongside `Add` and `Sort By`. Because, otherwise, if Sabaki isn't somehow configured, everytime you open `Manage Games`, you might accidentally see the previews. Again, that would be annoying. I would suggest for this to be another option inside the `Preferences`.
- A way of **referencing a game in the collection** through some sort of hyperlink.
    - Similar to what is already done through using `#<number_of_the_move>` to reference the inside a game.
- **(meta)Description** of a game collection.
    - Just as a game can have its comments and description, like players' names, dates, etc., game collections can too.
    - How and where to put this I haven't yet thought of.
    - Again, I don't know if that's possible within the limits of `sgf` standards.

If all or some of these features seem reasonable, perhaps it would be to later put them into separate issues. But it would be useful to have a single issue to serve as a hub for the bigger picture discussion.

**Comment by yishn** [comment]
> * **A title for each game in the collection**.

You can already give each game a name in the game info drawer. The name field value will be shown in the game collection manager.

>   * As a subfeature, there could also be a toggle to see the games in the collection as a list, where the titles would differentiate each one.

This is #146 

> * A **toggle button where you can set the preview to display the beginning of the games or their end**. [...] This spoiling can even be a nuissance for normal games, because sometimes you don't want to see the final position.

We don't show the end position per se. Just 30 moves into the game, so it wouldn't be a spoiler for normal games. But of course, for Tsumegos, that would be the end position. For normal games, displaying the beginning of the games would mostly just be a blank board which isn't helpful information at all. Maybe we can just check how long the game is instead of having an option.

**Comment by psygo** [comment]
> You can already give each game a name in the game info drawer. The name field value will be shown in the game collection manager.

You are correct. I hadn't noticed it. Nevertheless, I think it would also be intuitive and comfortable from a user's standpoint to have a way of renaming the game directly on the `Manage Games` preview. For example, this could be achieved by adding a `Rename Game` on the `Right Click` submenu.

> For normal games, displaying the beginning of the games would mostly just be a blank board which isn't helpful information at all.

That's one of the reasons why I suggested a *toggle button* or another *preference item*. If the user feels like the preview isn't adding any information because it's only showing the initial position, then he toggles it. 

I think it would be very difficult for the developer to sort of somehow guess which position number should be shown. It would be *far* easier if the user could choose what type of preview he wants.

**Comment by psygo** [comment]
By the way, it is a bit off-topic, but I would like to suggest another name for the label `enhancement`. Wouldn't something like `feature suggestion` be more accurate and specific? `enhancement` could mean that I'm just trying to improve the code for example.

I would also suggest adding brief descriptions to the labels so this type of "confusion" could be solved without you having to address it repeatedly.

**Comment by Terkwood** [comment]
Hope you don't mind a third party jumping in on a secondary topic, but I
like "enhancement" as a label, since it aligns with the default labels used
by github projects. I think the meaning is clear. Cheers.

**Comment by yishn** [comment]
I'm keeping the "enhancement" label. Please stop posting off-topic comments.
